crates/mongodb-agent-common/src/mongodb/selection.rs

```diff
@@ -161,33 +158,43 @@ fn selection_for_field(parent_columns: &[&str], field: &Field) -> Result<Bson, M
 }
 
 fn selection_for_array(
-    parent_columns: &[&str],
+    parent: ColumnRef<'_>,
     field: &NestedField,
     array_nesting_level: usize,
 ) -> Result<Bson, MongoAgentError> {
     match field {
         NestedField::Object(NestedObject { fields }) => {
-            let nested_parent_col_path = format!("${}", parent_columns.join("."));
-            let mut nested_selection = from_query_request_helper(&["$this"], fields)?;
+            let mut nested_selection =
+                from_query_request_helper(Some(ColumnRef::variable("this")), fields)?;
             for _ in 0..array_nesting_level {
                 nested_selection = doc! {"$map": {"input": "$$this", "in": nested_selection}}
             }
-            let map_expression =
-                doc! {"$map": {"input": &nested_parent_col_path, "in": nested_selection}};
-            Ok(doc! {"$cond": {"if": &nested_parent_col_path, "then": map_expression, "else": Bson::Null}}.into())
+            let map_expression = doc! {"$map": {"input": parent.clone().into_aggregate_expression(), "in": nested_selection}};
+            Ok(doc! {"$cond": {"if": parent.into_aggregate_expression(), "then": map_expression, "else": Bson::Null}}.into())
         }
         NestedField::Array(NestedArray {
             fields: nested_field,
-        }) => selection_for_array(parent_columns, nested_field, array_nesting_level + 1),
+        }) => selection_for_array(parent, nested_field, array_nesting_level + 1),
     }
 }
-fn append_to_path<'a, 'b, 'c>(parent_columns: &'a [&'b str], column: &'c str) -> Vec<&'c str>
-where
-    'b: 'c,
-{
-    parent_columns.iter().copied().chain(Some(column)).collect()
+
+fn nested_column_reference<'a>(
+    parent: Option<ColumnRef<'a>>,
+    column: &'a FieldName,
+) -> ColumnRef<'a> {
+    match parent {
+        Some(parent) => parent.into_nested_field(column),
+        None => ColumnRef::from_field_path([column]),
+    }
 }
 
+// fn append_to_path<'a, 'b, 'c, T>(parent_columns: &'a [&'b T], column: &'c T) -> Vec<&'c T>
+// where
+//     'b: 'c,
+// {
+//     parent_columns.iter().copied().chain(Some(column)).collect()
+// }
+
```

Comment:
Did you leave this intentionally or can it be cleaned up?

Reply:
Thanks for catching that - I removed it